### PR TITLE
[Feature][Connector] add starrocks save_mode

### PR DESCRIPTION
**Pull request by chl-wxp**
### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by hailin0 — COMMENTED** [review]
Please update the documentation

**Comment by chl-wxp** [comment]
> Please update the documentation

added

File: seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/catalog/StarRocksCatalog.java [hunk 1]
```diff
@@ -208,13 +211,70 @@ public CatalogTable getTable(TablePath tablePath)
...
+            }
+        } catch (Exception e) {
+            throw new CatalogException(String.format("Failed EXECUTE SQL in catalog %s", sql), e);
+        }
+    }
+
+    public boolean isExistsData(TablePath tablePath) {
+        try (Connection connection = DriverManager.getConnection(defaultUrl, username, pwd)) {
+            String sql = String.format("select * from %s limit 1", tablePath.getFullName());
+            PreparedStatement ps = connection.prepareStatement(sql);
```
Comment (Carl-Zhou-CN) [inline seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/catalog/StarRocksCatalog.java]: It seems that the method of obtaining a Statement is different in each place. Can you write it in the same way

File: seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/config/StarRocksSinkOptions.java [hunk 2]
```diff
@@ -131,10 +129,15 @@ public interface StarRocksSinkOptions {
...
                     .withDescription("");
-
-    SingleChoiceOption<DataSaveMode> SAVE_MODE =
-            Options.key(SupportSaveMode.DATA_SAVE_MODE_KEY)
-                    .singleChoice(DataSaveMode.class, Arrays.asList(DataSaveMode.APPEND_DATA))
+    Option<SchemaSaveMode> SCHEMA_SAVE_MODE =
+            Options.key("schema_save_mode")
+                    .enumType(SchemaSaveMode.class)
+                    .defaultValue(SchemaSaveMode.CREATE_SCHEMA_WHEN_NOT_EXIST)
+                    .withDescription("schema_save_mode");
```
Comment (Carl-Zhou-CN) [inline seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/config/StarRocksSinkOptions.java]: I hope there is a more detailed description

File: seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/config/StarRocksSinkOptions.java [hunk 3]
```diff
@@ -144,4 +147,7 @@ public interface StarRocksSinkOptions {
                     .intType()
                     .defaultValue(3 * 60 * 1000)
                     .withDescription("Set http socket timeout, default is 3 minutes.");
+
+    Option<String> CUSTOM_SQL =
+            Options.key("custom_sql").stringType().noDefaultValue().withDescription("custom_sql");
```
Comment (Carl-Zhou-CN) [inline seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/config/StarRocksSinkOptions.java]: Same as above

File: docs/en/connector-v2/sink/StarRocks.md [hunk 4]
```diff
@@ -20,31 +20,68 @@ The internal implementation of StarRocks sink connector is cached and imported b
 
 ## Sink Options
```
Comment (Carl-Zhou-CN) [inline docs/en/connector-v2/sink/StarRocks.md:21]: I think an example needs to be added to the documentation

Comment (chl-wxp) [reply docs/en/connector-v2/sink/StarRocks.md:21]: added

File: seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/config/StarRocksSinkOptions.java (hunk 2 above)
Comment (EricJoy2048) [reply seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/config/StarRocksSinkOptions.java]: > I hope there is a more detailed description

Yes, only copy the option name to here doesn't make any sense.

File: seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/config/StarRocksSinkOptions.java (hunk 3 above)
Comment (chl-wxp) [reply seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/config/StarRocksSinkOptions.java]: updated

File: seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/config/StarRocksSinkOptions.java (hunk 2 above)
Comment (chl-wxp) [reply seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/config/StarRocksSinkOptions.java]: updated

File: seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/catalog/StarRocksCatalog.java (hunk 1 above)
Comment (chl-wxp) [reply seatunnel-connectors-v2/connector-starrocks/src/main/java/org/apache/seatunnel/connectors/seatunnel/starrocks/catalog/StarRocksCatalog.java]: uptaded

**Review by hailin0 — COMMENTED** [review]
why not remove this ddl and use savemode？

StarRocksIT.java
```java
private static final String DDL_SINK = ...
```

https://github.com/apache/seatunnel/pull/6029/files#diff-40a4a1be8ee6ae052ddce2c338818c1f08199daf089ede3e82cc5a351a2521d6R66

**Comment by chl-wxp** [comment]
> why not remove this ddl and use savemode？
> 
> StarRocksIT.java
> 
> ```java
> private static final String DDL_SINK = ...
> ```
> 
> https://github.com/apache/seatunnel/pull/6029/files#diff-40a4a1be8ee6ae052ddce2c338818c1f08199daf089ede3e82cc5a351a2521d6R66

updated

**Comment by chl-wxp** [comment]
> why not remove this ddl and use savemode？
> 
> StarRocksIT.java
> 
> ```java
> private static final String DDL_SINK = ...
> ```
> 
> https://github.com/apache/seatunnel/pull/6029/files#diff-40a4a1be8ee6ae052ddce2c338818c1f08199daf089ede3e82cc5a351a2521d6R66

image

**Comment by chl-wxp** [comment]
> why not remove this ddl and use savemode？
> 
> StarRocksIT.java
> 
> ```java
> private static final String DDL_SINK = ...
> ```
> 
> https://github.com/apache/seatunnel/pull/6029/files#diff-40a4a1be8ee6ae052ddce2c338818c1f08199daf089ede3e82cc5a351a2521d6R66

deleted

**Review by ic4y — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**